From: "Bernardo C. Gutierrez Cantu" <bercantu@amazon.de>
Subject: mm: memblock: fix arguments passed to memblock_set_node()
Date: Fri, 25 Apr 2025 10:20:03 +0000

memblock_set_node() receives a `base` and a `size` arguments, but we are
passing the `start` and `end` of the memory regions when iterating over
them in memmap_init_reserved_pages() to set their node ids.

This results in the function setting the node ids for the reserved memory
regions in `[base, base + base + size)` instead of `[base, base + size)`.

Pass `start` and `size`, so that we iterate over the correct range.

Link: https://lkml.kernel.org/r/20250425102003.64122-1-bercantu@amazon.de
Fixes: 61167ad5fecd ("mm: pass nid to reserve_bootmem_region()")
Signed-off-by: Bernardo C. Gutierrez Cantu <bercantu@amazon.de>
Acked-by: David Woodhouse <dwmw2@infradead.org>
Cc: Mike Rapoport <rppt@kernel.org>
Cc: Yajun Deng <yajun.deng@linux.dev>
Cc: Huacai Chen <chenhuacai@kernel.org>
Cc: Huang Pei <huangpei@loongson.cn>
Cc: Jiaxun Yang <jiaxun.yang@flygoat.com>
Cc: Thomas Bogendoerfer <tsbogend@alpha.franken.de>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/memblock.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/mm/memblock.c~mm-memblock-fix-arguments-passed-to-memblock_set_node
+++ a/mm/memblock.c
@@ -2196,7 +2196,7 @@ static void __init memmap_init_reserved_
 		if (memblock_is_nomap(region))
 			reserve_bootmem_region(start, end, nid);
 
-		memblock_set_node(start, end, &memblock.reserved, nid);
+		memblock_set_node(start, region->size, &memblock.reserved, nid);
 	}
 
 	/*
_
